aci/resource_aci_dhcprelayp.go

```diff
@@ -0,0 +1,339 @@
+					"infra",
+					"tenant",
+				}, false),
+			},
+
+			"relation_dhcp_rs_prov": &schema.Schema{
+				Type:     schema.TypeSet,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Optional: true,
+				Set:      schema.HashString,
+			},
+		}),
+	}
+}
+func getRemoteDHCPRelayPolicy(client *client.Client, dn string) (*models.DHCPRelayPolicy, error) {
+	dhcpRelayPCont, err := client.Get(dn)
+	if err != nil {
+		return nil, err
+	}
+
+	dhcpRelayP := models.DHCPRelayPolicyFromContainer(dhcpRelayPCont)
+
+	if dhcpRelayP.DistinguishedName == "" {
+		return nil, fmt.Errorf("DHCPRelayPolicy %s not found", dhcpRelayP.DistinguishedName)
+	}
+
+	return dhcpRelayP, nil
+}
+
+func setDHCPRelayPolicyAttributes(dhcpRelayP *models.DHCPRelayPolicy, d *schema.ResourceData) *schema.ResourceData {
+	dn := d.Id()
+
+	d.SetId(dhcpRelayP.DistinguishedName)
+	d.Set("description", dhcpRelayP.Description)
+
+	if dn != dhcpRelayP.DistinguishedName {
+		d.Set("tenant_dn", "")
+	}
+	dhcpRelayPMap, _ := dhcpRelayP.ToMap()
+
+	d.Set("name", dhcpRelayPMap["name"])
+
+	d.Set("annotation", dhcpRelayPMap["annotation"])
+	d.Set("mode", dhcpRelayPMap["mode"])
+	d.Set("name_alias", dhcpRelayPMap["nameAlias"])
+	d.Set("owner", dhcpRelayPMap["owner"])
+	return d
+}
+
+func resourceAciDHCPRelayPolicyImport(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	log.Printf("[DEBUG] %s: Beginning Import", d.Id())
+	aciClient := m.(*client.Client)
+
+	dn := d.Id()
+
+	dhcpRelayP, err := getRemoteDHCPRelayPolicy(aciClient, dn)
+
+	if err != nil {
+		return nil, err
+	}
+	schemaFilled := setDHCPRelayPolicyAttributes(dhcpRelayP, d)
+
+	log.Printf("[DEBUG] %s: Import finished successfully", d.Id())
+
+	return []*schema.ResourceData{schemaFilled}, nil
+}
+
+func resourceAciDHCPRelayPolicyCreate(d *schema.ResourceData, m interface{}) error {
+	log.Printf("[DEBUG] DHCPRelayPolicy: Beginning Creation")
+	aciClient := m.(*client.Client)
+	desc := d.Get("description").(string)
+
+	name := d.Get("name").(string)
+
+	TenantDn := d.Get("tenant_dn").(string)
+
+	dhcpRelayPAttr := models.DHCPRelayPolicyAttributes{}
+	if Annotation, ok := d.GetOk("annotation"); ok {
+		dhcpRelayPAttr.Annotation = Annotation.(string)
+	} else {
+		dhcpRelayPAttr.Annotation = "{}"
+	}
+	if Mode, ok := d.GetOk("mode"); ok {
+		dhcpRelayPAttr.Mode = Mode.(string)
+	}
+	if NameAlias, ok := d.GetOk("name_alias"); ok {
+		dhcpRelayPAttr.NameAlias = NameAlias.(string)
+	}
+	if Owner, ok := d.GetOk("owner"); ok {
+		dhcpRelayPAttr.Owner = Owner.(string)
+	}
+	dhcpRelayP := models.NewDHCPRelayPolicy(fmt.Sprintf("relayp-%s", name), TenantDn, desc, dhcpRelayPAttr)
+
+	err := aciClient.Save(dhcpRelayP)
+	if err != nil {
+		return err
+	}
+	d.Partial(true)
+
+	d.SetPartial("name")
+
+	d.Partial(false)
+
+	checkDns := make([]string, 0, 1)
+
+	if relationTodhcpRsProv, ok := d.GetOk("relation_dhcp_rs_prov"); ok {
+		relationParam := relationTodhcpRsProv.(string)
```

Comment:
This relation has list of dns. So update the logic here for accepting multiple strings.